### PR TITLE
feat(route):增加leetcode每日一题题解

### PR DESCRIPTION
**Pull request by woaidouya123**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/leetcode/today
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/leetcode/today.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        Origin: `https://leetcode-cn.com`,
+    };
+    // 获取每日一题
+    const data = (
+        await got({
+            method: 'post',
+            url,
+            json: {
+                operationName: 'questionOfToday',
+                query: '\n    query questionOfToday {\n  todayRecord {\n    date\n    userStatus\n    question {\n      questionId\n      frontendQuestionId: questionFrontendId\n      difficulty\n      title\n      titleCn: translatedTitle\n      titleSlug\n      paidOnly: isPaidOnly\n      freqBar\n      isFavor\n      acRate\n      status\n      solutionNum\n      hasVideoSolution\n      topicTags {\n        name\n        nameTranslated: translatedName\n        id\n      }\n      extra {\n        topCompanyTags {\n          imgUrl\n          slug\n          numSubscribed\n        }\n      }\n    }\n    lastSubmission {\n      id\n    }\n  }\n}\n    ',
```
Comment (TonyRL) [inline lib/v2/leetcode/today.js:17]: Please use [Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) with natural linebreaks.

https://github.com/DIYgod/RSSHub/blob/aef0b0a9b086ea55a4cfb5b74522d4935c836d3d/lib/v2/leetcode/dailyquestion-cn.js#L19-L27

File: lib/v2/leetcode/today.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
...lug: String!) {\n  question(titleSlug: $titleSlug) {\n    questionId\n    questionFrontendId\n    categoryTitle\n    boundTopicId\n    title\n    titleSlug\n    content\n    translatedTitle\n    translatedContent\n    isPaidOnly\n    difficulty\n    likes\n    dislikes\n    isLiked\n    similarQuestions\n    contributors {\n      username\n      profileUrl\n      avatarUrl\n      __typename\n    }\n    langToValidPlayground\n    topicTags {\n      name\n      slug\n      translatedName\n      __typename\n    }\n    companyTagStats\n    codeSnippets {\n      lang\n      langSlug\n      code\n      __typename\n    }\n    stats\n    hints\n    solution {\n      id\n      canSeeDetail\n      __typename\n    }\n    status\n    sampleTestCase\n    metaData\n    judgerAvailable\n    judgeType\n    mysqlSchemas\n    enableRunCode\n    envInfo\n    book {\n      id\n      bookName\n      pressName\n      source\n      shortDescription\n      fullDescription\n      bookImgUrl\n      pressImgUrl\n      productUrl\n      __typename\n    }\n    isSubscribed\n    isDailyQuestion\n    dailyRecordStatus\n    editorType\n    ugcQuestionId\n    style\n    exampleTestcases\n    __typename\n  }\n}\n',
```
Comment (TonyRL) [inline lib/v2/leetcode/today.js:33]: Please use [Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) with natural linebreaks.

https://github.com/DIYgod/RSSHub/blob/aef0b0a9b086ea55a4cfb5b74522d4935c836d3d/lib/v2/leetcode/dailyquestion-cn.js#L19-L27

File: lib/v2/leetcode/today.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
...Articles($questionSlug: String!, $skip: Int, $first: Int, $orderBy: SolutionArticleOrderBy, $userInput: String, $tagSlugs: [String!]) {\n  questionSolutionArticles(questionSlug: $questionSlug, skip: $skip, first: $first, orderBy: $orderBy, userInput: $userInput, tagSlugs: $tagSlugs) {\n    totalNum\n    edges {\n      node {\n        ...solutionArticle\n        __typename\n      }\n      __typename\n    }\n    __typename\n  }\n}\n\nfragment solutionArticle on SolutionArticleNode {\n  rewardEnabled\n  canEditReward\n  uuid\n  title\n  slug\n  sunk\n  chargeType\n  status\n  identifier\n  canEdit\n  canSee\n  reactionType\n  reactionsV2 {\n    count\n    reactionType\n    __typename\n  }\n  tags {\n    name\n    nameTranslated\n    slug\n    tagType\n    __typename\n  }\n  createdAt\n  thumbnail\n  author {\n    username\n    profile {\n      userAvatar\n      userSlug\n      realName\n      __typename\n    }\n    __typename\n  }\n  summary\n  topic {\n    id\n    commentCount\n    viewCount\n    __typename\n  }\n  byLeetcode\n  isMyFavorite\n  isMostPopular\n  isEditorsPick\n  hitCount\n  videosInfo {\n    videoId\n    coverUrl\n    duration\n    __typename\n  }\n  __typename\n}\n',
```
Comment (TonyRL) [inline lib/v2/leetcode/today.js:48]: Please use [Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) with natural linebreaks.

https://github.com/DIYgod/RSSHub/blob/aef0b0a9b086ea55a4cfb5b74522d4935c836d3d/lib/v2/leetcode/dailyquestion-cn.js#L19-L27

File: lib/v2/leetcode/today.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                        query: 'query solutionDetailArticle($slug: String!, $orderBy: SolutionArticleOrderBy!) {\n  solutionArticle(slug: $slug, orderBy: $orderBy) {\n    ...solutionArticle\n    content\n    question {\n      questionTitleSlug\n      __typename\n    }\n    position\n    next {\n      slug\n      title\n      __typename\n    }\n    prev {\n      slug\n      title\n      __typename\n    }\n    __typename\n  }\n}\n\nfragment solutionArticle on SolutionArticleNode {\n  rewardEnabled\n  canEditReward\n  uuid\n  title\n  slug\n  sunk\n  chargeType\n  status\n  identifier\n  canEdit\n  canSee\n  reactionType\n  reactionsV2 {\n    count\n    reactionType\n    __typename\n  }\n  tags {\n    name\n    nameTranslated\n    slug\n    tagType\n    __typename\n  }\n  createdAt\n  thumbnail\n  author {\n    username\n    profile {\n      userAvatar\n      userSlug\n      realName\n      __typename\n    }\n    __typename\n  }\n  summary\n  topic {\n    id\n    commentCount\n    viewCount\n    __typename\n  }\n  byLeetcode\n  isMyFavorite\n  isMostPopular\n  isEditorsPick\n  hitCount\n  videosInfo {\n    videoId\n    coverUrl\n    duration\n    __typename\n  }\n  __typename\n}\n',
```
Comment (TonyRL) [inline lib/v2/leetcode/today.js:68]: Please use [Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) with natural linebreaks.

https://github.com/DIYgod/RSSHub/blob/aef0b0a9b086ea55a4cfb5b74522d4935c836d3d/lib/v2/leetcode/dailyquestion-cn.js#L19-L27

File: lib/v2/leetcode/today.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    ctx.state.data = {
+        title: 'LeetCode Today',
+        description: 'LeetCode 每日一题',
+        link: questionUrl,
+        item: [
+            {
+                title: `每日一题-${question.translatedTitle}`,
+                link: questionUrl,
+                description: question.translatedContent,
+                pubDate: data.todayRecord[0].date + ' 00:00:00',
```
Comment (TonyRL) [inline lib/v2/leetcode/today.js:94]:
```suggestion
                pubDate: parseDate(data.todayRecord[0].date),
```
https://docs.rsshub.app/joinus/pub-date.html

File: lib/v2/leetcode/today.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            },
+            ...articleContent.map((art, i) => ({
+                title: art.title,
+                link: `${questionUrl}/solution/${art.slug}`,
+                description: converter.makeHtml(handleText(art.content)),
+                pubDate: articles[i].node.createdAt,
```
Comment (TonyRL) [inline lib/v2/leetcode/today.js:100]:
```suggestion
                pubDate: parseDate(articles[i].node.createdAt),
```
https://docs.rsshub.app/joinus/pub-date.html

File: yarn.lock [hunk 7]
```diff
@@ -11680,10 +11680,10 @@ ramda@^0.26.1:
   resolved "https://registry.yarnpkg.com/ramda/-/ramda-0.26.1.tgz#8d41351eb8111c55353617fc3bbffad8e4d35d06"
   integrity sha512-hLWjpy7EnsDBb0p+Z3B7rPi3GDeRG5ZtiI33kJhTt+ORCd38AbAIjB/9zRIUoeTbE/AVX5ZkU7m6bznsvrf8eQ==
 
-rand-user-agent@1.0.76:
-  version "1.0.76"
-  resolved "https://registry.yarnpkg.com/rand-user-agent/-/rand-user-agent-1.0.76.tgz#af5e8a8f838bae5d149842d7a5633704935b0ce1"
-  integrity sha512-xQSfN3uwFHbzEolpniv20mICB7v9HS8Xz+xo3ixSxRXokNJUh1zF7wVJz/8wJovItiG9+3zput65+erPyDPJsQ==
+rand-user-agent@1.0.77:
```
Comment (TonyRL) [inline yarn.lock:11683]: Please remove any changes on `yarn.lock` and `package.json` unless you're adding new package(s)

Comment (woaidouya123) [reply yarn.lock:11683]: Thanks for the comments, I'll fix it later.

**Pull request closed without merging**